transform/src/Components/TransformStep.tsx

```diff
@@ -0,0 +1,61 @@
+import {Expression} from './Expression';
+import {ActionGroup, Button, Form } from '@patternfly/react-core';
+import {useState} from 'react';
+
+type TransformStepParams = {
+  simple?: string;
+  jq?: string;
+};
+
+export const TransformStep = (props: any) => {
+  let initSyntax;
+  let initExpression;
+  if (props.stepParams?.jq) {
+    initSyntax = 'jq';
+    initExpression = props.stepParams?.jq || '';
+  } else {
+    initSyntax = 'simple';
+    initExpression = props.stepParams?.simple || '';
+  }
+
+  const [stepParams, setStepParams] = useState<TransformStepParams>({
+        simple: props.stepParams?.simple,
+        jq: props.stepParams?.jq,
+    });
+
+  function updateKaoto() {
+    props.notifyKaoto('Transform step updated');
+    props.updateStepParams(stepParams);
+  }
+
+  function updateStepParams(simple?: string, jq?: string) {
+    const newStepParams: TransformStepParams = {
+      simple: simple ? simple : undefined,
```

Comment:
You can probably just change this from `simple: simple ? simple : undefined` to `simple` and it will be resolved to either the value or `undefined`. This is object property shorthand from ES6.